src/spikeinterface/sorters/launcher.py

```diff
@@ -4,69 +4,187 @@
 from pathlib import Path
 import shutil
 import numpy as np
-import json
 import tempfile
 import os
 import stat
 import subprocess
 import sys
+import warnings
 
-from spikeinterface.core import load_extractor, aggregate_units
-from spikeinterface.core.core_tools import check_json
+from spikeinterface.core import  aggregate_units
 
 from .sorterlist import sorter_dict
-from .runsorter import run_sorter, run_sorter
-
-
-def _run_one(arg_list):
-    # the multiprocessing python module force to have one unique tuple argument
-    (
-        sorter_name,
-        recording,
-        output_folder,
-        verbose,
-        sorter_params,
-        docker_image,
-        singularity_image,
-        with_output,
-    ) = arg_list
-
-    if isinstance(recording, dict):
-        recording = load_extractor(recording)
+from .runsorter import run_sorter
+from .basesorter import is_log_ok
+
+_default_engine_kwargs = dict(
+    loop=dict(),
+    joblib=dict(n_jobs=-1, backend="loky"),
+    processpoolexecutor=dict(max_workers=2, mp_context=None),
+    dask=dict(client=None),
+    slurm=dict(tmp_script_folder=None, cpus_per_task=1, mem="1G"),
+)
+
+
+_implemented_engine = list(_default_engine_kwargs.keys())
+
+def run_sorter_jobs(job_list, engine="loop", engine_kwargs={}, return_output=False):
+    """
+    Run several :py:func:`run_sorter()` sequencially or in parralel given a list of job.
+
+    For **engine="loop"** this is equivalent to:
+
+    ..code::
+
+        for job in job_list:
+            run_sorter(**job)
+    
+    For some engines, this function is blocking until the results ("loop", "joblib", "multiprocessing", "dask").
```

Comment:
I'm also not sure the meaning of this line as written

Reply:
Oups. Any idea how reformulate this ?
blocking is when the function call wait until the result (asynchronous call)
oposite to asynchronous (return immediatly)

Reply:
I see then I would just say: 

For some engines (...) this function is blocking until the results are returned.
For other engines (..) the function returns almost immediately (aka non-blocking), but the results must be retrieved by hand when the :py:func:`read_sorter_folder()` has finished.

Using the 'but' emphasizes that although the call is asynchronous it is up to the user to get the results when they are ready. I think the confusion is that the protected nature of the word `return` can be tricky to users not used to asyn vs syn programming.

Reply:
My suggestions:
1) This should be added the Notes section as the numpy docstring convention:

https://numpydoc.readthedocs.io/en/latest/format.html#notes

2) As for the clarification, I think lists are better in case you want to add notes later. 
```pythoh
"""
The following engines block the I/O:
* Item1
* Item2
* Item3

Where *blocking* means that the results are {you description of what bocking is from the point of view of the user} . This is in opposition to *asynchronous* wher the {your description of what the antonym is from the point of view of the user}.
 """
```
This later formulation has the advantage that it separates the information of which items are blocking from the description / defintion that you give to help users who don't understand the term.  Advanced users can stop reading as they already know the following information but those who don't can keep reading after the list (this is the same convention that mathematical formulas follow where a description is added after a formula is presented). Plus, as @zm711 suggested you define something by what it is (giving a googlable term) and then also by what is not (negative definition) giving some much needed redundancy for first time readers.

Reply:
That's a great strategy from @h-mayorquin. I think the issue is that since users will directly interface with this function the bilayer approach of giving the advanced person an off-ramp because they understand and the beginner person enough information so they can at least google later is super important.